src/abstract-interpretation/data-frame/absint-visitor.ts

```diff
@@ -57,22 +57,25 @@ class DataFrameAbsintVisitor<
 		const predecessors = this.getPredecessorNodes(vertex.id);
 		this.newDomain = joinDataFrameStates(...predecessors.map(node => node.info.dataFrame?.domain ?? new Map()));
 		this.onVisitNode(nodeId);
+
 		const visitedCount = this.visited.get(vertex.id) ?? 0;
 		this.visited.set(vertex.id, visitedCount + 1);
 
-		if(visitedCount >= (this.config.wideningThreshold ?? DefaultWideningThreshold)) {
-			this.newDomain = wideningDataFrameStates(this.oldDomain, this.newDomain);
-		}
 		return visitedCount === 0 || !equalDataFrameState(this.oldDomain, this.newDomain);
 	}
-	
+
 	protected override visitDataflowNode(vertex: Exclude<CfgSimpleVertex, CfgBasicBlockVertex>): void {
 		const node = this.getNormalizedAst(isMarkerVertex(vertex) ? vertex.root : vertex.id);
 
 		if(node !== undefined) {
 			this.oldDomain = node.info.dataFrame?.domain ?? new Map<NodeId, DataFrameDomain>();
 			super.visitDataflowNode(vertex);
-			node.info.dataFrame ??= {};
+
+			if((this.visited.get(vertex.id) ?? 0) >= (this.config.wideningThreshold ?? DefaultWideningThreshold)) {
+				this.newDomain = wideningDataFrameStates(this.oldDomain, this.newDomain);
+			}
+			// mark variable definitions as "unassigned", since the evaluation of the assigned value is delayed until processing the assignment
+			node.info.dataFrame ??= isVariableDefinitionVertex(this.getDataflowGraph(vertex.id)) ? { type: 'unassigned' } : {};
```

Comment:
Conceptually, shouldn't we stop here if widened? I mean yes we can continue in the future, but it feels, weird to make a default unassigned set if widening applies